ScienceDirect.js

```diff
@@ -108,15 +112,58 @@ function getPDFLink(doc, onDone) {
 		}
 	}
 	
-	// If none of that worked for some reason, get the URL from the initial HTML, where it is present,
-	// by fetching the page source again. Hopefully this is never actually used.
+	// On some institutional networks with access to ScienceDirect, the site
+	// serves JSON metadata probably used to preload dynamic content without a
+	// separate network request. If we find it, we can take advantage of it to
+	// grab the PDF url's parts directly, constructing it in the same way that
+	// the site's frontend JavaScript would.
+	var json = doc.querySelector('script[type="application/json"]').innerHTML;
```

Comment:
We should be a bit more defensive just in case there's ever not a `<script type="application/json">` block at all, which would cause this to throw.